### ISSUE TITLE
Train the Ganomaly model using a self-made dataset, and the labels of the test results are all anomalous[Task]: 

### ISSUE THREAD

**Issue by MMYY-yy** [openvinotoolkit/anomalib#1624]
### What is the motivation for this task?

1.Whether adjusting parameters or increasing the number of positive samples for training, the loss is around 20
2. Train using the official Bottle dataset, and test to detect that all labels are anomalous

![000](https://github.com/openvinotoolkit/anomalib/assets/156411952/c3946b55-ea16-405d-b314-9ce86c512ed7)

### Describe the solution you'd like

dataset:
  name: SY1  # 数据集的名称,这个不重要
  format: folder
  path: ./data/SY1  # 自制数据集的地址
  normal_dir: SY1_OK_train  # 自制数据集正样本子文件夹
  abnormal_dir: SY1_NG_test  # 自制数据集负样本子文件夹
  normal_test_dir: SY1_OK_test  # 包含正常测试图像的文件夹的名称
  mask_dir: null   # 二值掩膜路径，自制数据集一般没有，填null
  task: classification  # classification or segmentation
  extensions: null
  category: OK
  train_batch_size: 16
  eval_batch_size: 16
  num_workers: 8
  image_size: [300, 700]  # dimensions to which images are resized (mandatory)
#  center_crop: 224 # dimensions to which images are center-cropped after resizing (optional)
  normalization: imagenet # data distribution to which the images will be normalized: [none, imagenet]
  transform_config:
    train: null
    eval: null
  test_split_mode: from_dir # options: [from_dir, synthetic]
  test_split_ratio: 0.2 # fraction of train images held out testing (usage depends on test_split_mode)
  val_split_mode: same_as_test # options: [same_as_test, from_test, synthetic]
  val_split_ratio: 0.5 # fraction of train/test images held out for validation (usage depends on val_split_mode)
  tiling:
    apply: false
    tile_size: null
    stride: null
    remove_border_count: 0
    use_random_tiling: False
    random_tile_count: 16

model:
  name: patchcore
  backbone: wide_resnet50_2
  pre_trained: true
  layers:
    - layer2
    - layer3
  coreset_sampling_ratio: 0.1
  num_neighbors: 9
  normalization_method: min_max # options: [null, min_max, cdf]

metrics:
  image:
    - F1Score
    - AUROC
#  pixel:     # 分类数据集没有ground truth的二值化掩膜（mask），因此需要将task字段设置为classification而不是segmentation。同时将metrics部分的pixel部分注释掉
#    - F1Score
#    - AUROC
  threshold:
    method: adaptive #options: [adaptive, manual]
    manual_image: null
    manual_pixel: null

visualization:
  show_images: False # show images on the screen
  save_images: True # save images to the file system
  log_images: True # log images to the available loggers (if any)
  image_save_path: null # path to which images will be saved
  mode: full # options: ["full", "simple"]

project:
  seed: 0
  path: ./results

logging:
  logger: [] # options: [comet, tensorboard, wandb, csv] or combinations.
  log_graph: false # Logs the model graph to respective logger.

optimization:
  export_mode: onnx # options: onnx, openvino

# PL Trainer Args. Don't add extra parameter here.
trainer:
  enable_checkpointing: true
  default_root_dir: null
  gradient_clip_val: 0
  gradient_clip_algorithm: norm
  num_nodes: 1
  devices: 1
  enable_progress_bar: true
  overfit_batches: 0.0
  track_grad_norm: -1
  check_val_every_n_epoch: 1 # Don't validate before extracting features.
  fast_dev_run: false
  accumulate_grad_batches: 1
  max_epochs: 1
  min_epochs: null
  max_steps: -1
  min_steps: null
  max_time: null
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  limit_test_batches: 1.0
  limit_predict_batches: 1.0
  val_check_interval: 1.0 # Don't validate before extracting features.
  log_every_n_steps: 50
  accelerator: auto # <"cpu", "gpu", "tpu", "ipu", "hpu", "auto">
  strategy: null
  sync_batchnorm: false
  precision: 32
  enable_model_summary: true
  num_sanity_val_steps: 0
  profiler: null
  benchmark: false
  deterministic: false
  reload_dataloaders_every_n_epochs: 0
  auto_lr_find: false
  replace_sampler_ddp: true
  detect_anomaly: false
  auto_scale_batch_size: false
  plugins: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle

**Comment by samet-akcay** [comment]
Duplicate of #1548, which we will be addressing later on.

**Comment by MMYY-yy** [comment]
I am using version V0.7.0 and encountered the following issues while using the Hyperparameter Optimization feature. May I ask what is the cause of this problem and how should it be resolved

“Optimizer failed to retrieve data from the optimizer service, for more information look at: https://comet.com/docs/v2/api-and-sdk/python-sdk/warnings-errors/”

---- Replied Message ----
| From | Samet ***@***.***> |
| Date | 01/12/2024 16:59 |
| To | openvinotoolkit/anomalib ***@***.***> |
| Cc | MMYY-yy ***@***.***>,
Author ***@***.***> |
| Subject | Re: [openvinotoolkit/anomalib] Train the Ganomaly model using a self-made dataset, and the labels of the test results are all anomalous[Task]: (Issue #1624) |

Closed #1624 as completed.
